networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java

```diff
@@ -317,6 +322,33 @@ public SafeFuture<Void> requestBlobsSidecarsByRange(
                 new UnsupportedOperationException("BlobsSidecarsByRange method is not available")));
   }
 
+  @Override
+  public SafeFuture<Void> requestBlobSidecarsByRange(
+      final UInt64 startSlot, final UInt64 count, final RpcResponseListener<BlobSidecar> listener) {
+    return rpcMethods
+        .blobSidecarsByRange()
+        .map(
+            method -> {
+              final UInt64 firstSupportedSlot = firstSlotSupportingBlobsSidecarsByRange.get();
+              final BlobSidecarsByRangeRequestMessage request;
+              if (startSlot.isLessThan(firstSupportedSlot)) {
+                LOG.debug(
+                    "Requesting blob sidecars from slot {} instead of slot {} because the request is spanning the Deneb fork transition",
+                    firstSupportedSlot,
+                    startSlot);
+                final UInt64 updatedCount =
+                    count.minusMinZero(firstSupportedSlot.minusMinZero(startSlot));
+                request = new BlobSidecarsByRangeRequestMessage(firstSupportedSlot, updatedCount);
```

Comment:
What if `updatedCount` is 0? Is it possible? Will we send a request? Some peer could penalize for such behavior, we'd better not

Reply:
Actually it could be 0 and it was a potential bug (the `getMaxSlot` in `BlobsSidecarsByRangeRequestMessage` and `BlobSidecarsByRangeRequestMessage` could throw an `ArithmeticException`).
Fixed that and added some tests.
Good catch!